code/datums/components/cleaning.dm

```diff
@@ -0,0 +1,36 @@
+/datum/component/cleaning
+
+/datum/component/cleaning/Initialize()
+	RegisterSignal(list(COMSIG_MOVABLE_MOVED), .proc/Clean)
+
+/datum/component/cleaning/proc/Clean()
+	var/atom/movable/AM = parent
+	if(!istype(AM))
+		return
```

Comment:
This should be checked in Initialize:

```dm
if(!ismovable(parent))
    . = COMPONENT_INCOMPATIBLE
    CRASH("[type] added to a [parent.type]")
```

Reply:
In OnTransfer() too

Reply:
Actually, hold off on that, I'll just write a more generic system. It's about time

Reply:
just do what cyber originally said, if he doesn't finish by the time it's time to merge your pr he just has one more component to fix